text/0000-feature-metadata-doc-cfg.md

```diff
@@ -0,0 +1,334 @@
+- Feature Name: feature-metadata-doc-cfg
+- Start Date: 2023-04-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/3416)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC has three simple goals:
+
+1. Add a way to write feature descriptions in `Cargo.toml`, as well some other
+   simple feature attributes.
+2. Establish a way for Cargo or other build systems to easily pass advanced
+   configuration information to `rustdoc` (e.g., favicon or playground URLs from
+   `Cargo.toml`)
+3. Allow `rustdoc` to accept this information either from the CLI or from a new
+   `rustdoc.toml` file.
+
+The outcome is that `rustdoc` will gain the ability to document cargo features
+and get its configuration from `Cargo.toml`, without having any awareness of
+Cargo project structure itself. There will also be room to grow for more
+advanced configuration options.
+
+# Motivation
+
+[motivation]: #motivation
+
+Currently, `docs.rs` provides a basic view of available feature flags on a
+rather simple page: for example, [`tokio`]. It is helpful as a quick overview of
+available features, but it is not managed by `rustdoc` (i.e., is not available
+on local) and there is no way to specify a description or other useful
+information.
+
+The second problem is that `rustdoc` has some per-crate configuration settings,
+such as relevant URLs, that are awkward to define in Rust source files using
+attributes. It is expected that there may be further configuration options in
+the future, for specifying things like:
+
+1. Resource manifests (paths to assets, such as `KaTeX` for math rendering or
+   non-image files)
+2. Non-code informational pages (such as [`clap`'s derive information])
+
+This RFC provides a way to solve both problems: it specifies a way to add
+user-facing metadata to cargo features, and specifies how that and other
+information should be passed to `rustdoc`.
+
+[`tokio`]: https://docs.rs/crate/tokio/latest/features
+[`clap`'s derive information]: https://docs.rs/clap/4.2.2/clap/_derive/index.html
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Usage is simple: features will be able to be specified in a table (inline or
+separate) with the keys `doc`, `public`, `deprecated`, and `requires`. Sample
```

Comment:
I'm not seeing these attributes specified (schema, semantics)

Reply:
This has been elaborated upon after the split